DREAM/Optimization/tsgParticleSwarm.hpp

```diff
@@ -50,26 +50,43 @@
- * specified or inferred from input data. Each vector is associated with a particular particle's position \b particle_position,
- * velocity \b particle_velocity, and best known position \b best_particle_position from a previous invocation of the particle swarm
- * algorithm ParticleSwarm(). The last \b num_dimensions entries of \b best_particle position contain the position of the
- * overall best position in the swarm.
+ * \par class ParticleSwarmState
+ * An object of this class represents the components of a particle swarm and all key parts of the swarm can be read through member
+ * methods. Methods are also provided to initialize the swarm and modify a subset of its parts.
  *
- * \par Particle Swarm Cache
- * When the particle swarm state is modified by invoking the particle swarm algorithm ParticleSwarm() with a particular objective
- * function and domain, the results of invocation are stored in several cache variables prefixed by \b cache_. This is done
- * to speed up subsequent calls of ParticleSwarm() on this modified swarm state, under the assumption that the objective function
- * and domain from the previous call remains unchanged.
+ * \par Constructors and Copy/Move assignment
+ * All constructors require information about number of dimensions and particles. Methods are available to retrieve these two
+ * numbers. No default constructor is available for this class, but copy/move constructors and assignment operator= overloads are
+ * available. The number of dimensions and particles \b cannot be modified.
+ * - ParticleSwarmState()
+ * - getNumDimensions()
+ * - getNumParticles()
  *
- * \par
- * To apply the ParticleSwarm() algorithm on a swarm state with an objective function or domain that is different from its
- * last invocation of ParticleSwarm(), the cache must be cleared using the class method clearCache(). The behavior of the
- * particle swarm algorithm without doing this step is otherwise \a undefined.
+ * \par Set and Get Particle Positions and Velocities
+ * Each particle of the swarm is associated with a (possibly empty) position and velocity vector. When an optimization method is
+ * applied to the swarm, each particle is also associated with a 'best' position in terms of the objective function given to the
+ * optimization method. Methods are also available to determine if the positions, best positions, and velocities are
+ * empty (uninitialized) or non-empty (initialized).
+ * - getParticlePositions(), setParticlePositions()
+ * - getParticleVelocities(), setParticleVelocities()
+ * - getBestParticlePositions(), setBestParticlePositions()
+ * - getBestPosition()
+ * - clearBestParticles()
+ * - isPositionInitialized(), isVelocityInitialized(), isBestPositionInitialized()
+ *
+ * \par Initializing a Random Swarm
+ * Helper methods are available for initializing random particle positions and velocities.
+ * - initializeParticlesInsideBox()
+ *
```

Comment:
Add another paragraph that explicitly lists the method to get the computed optimum, i.e., best-position. Also, give some explanation of what that means, e.g., the optimal value observed throughout the entire series across all particles.

Reply:
Done. I've also made a separate paragraph for `getBestParticlePositions()` and its related functions.